extplane-plugin/xplaneplugin.cpp

```diff
@@ -329,9 +329,36 @@ bool XPlanePlugin::loadSituation(QString sitFileLocation) {
 
     // XPLMLoadDataFile's return value is not documented, assuming it returns
     // 1 on success and 0 on fail. TODO: Check this.
+
     return XPLMLoadDataFile(xplm_DataFile_Situation, sitFileLocation.toUtf8().data());
 }
 
+/**
+ * @brief XPlanePlugin::addFMSEntryLatLon
+ * @param fmsEntryLine - string in format "id,latitude,longitude,altitude"
+ * example: "1,50.0267,8.51,198"
+ */
+void XPlanePlugin::addFMSEntryLatLon(QString fmsEntryLine){
+     QStringList params = fmsEntryLine.split(",", QString::SkipEmptyParts);
+     int id = params.value(0).toInt();
```

Comment:
You could check for invalid values here before continuing.. But i guess it's ok, it'll just put zeros in place of invalid values given and this should not crash.

Reply:
Docs also say that "There is a max of 100 waypoints in the flight plan. " so please add a check to limit max number of entries to 100. Also looks like entry -1 would go through and could cause a crash. So check that id values are between 0 and 100.